backend/api/routes/user.py

```diff
@@ -336,3 +336,62 @@ async def get_company_ranking_history(company_id: str, db: Session = Depends(get
         models.Ranking.created_at.desc()).all()
 
     return rankings
+
+@router.get('/company/ranks/{category}', tags=["User"], )
+def get_company_category(category: str, user: User = Depends(get_current_user)):
+    """
+    This gets the metrics of a company by category for an authenticated user
+    """
+    db: Session = next(get_db())
+
+    # TODO: Validate and ensure the user has active subscription for a low cap company
+    is_user_subscribed = False
+
+    if category == low_cap_category_id and not is_user_subscribed:
+        raise HTTPException(status_code=401,
+                            detail="You must be subscribed before you can access low cap companies")
+    # get companies
+    companies: list = db.query(models.Company).filter(models.Company.category == category).all()
+
+    # get latest rankings
+    rankings: list = []
+    for company in companies:
+        rank = db.query(Ranking).filter(Ranking.company == company.company_id) \
+            .order_by(Ranking.created_at.desc()).first()
+        if rank:
+            rankings.append(rank)
+
+    # sort rankings by score (descending)
+    def get_ranking_sort_key(inner_rank: Ranking):
```

Comment:
what's this really. doing?

Reply:
The low_market_cap is limited to only authenticated and subscribed users, so this endpoint validates the user's subscription to make all stock categories available.